libminifi/include/core/state/Value.h

```diff
@@ -86,35 +87,57 @@ class Value {
   }
 
   virtual bool getValue(uint32_t &ref) {
-    const auto negative = string_value.find_first_of('-') != std::string::npos;
-     if (negative) {
-       return false;
-     }
-    ref = std::stoul(string_value);
+    try {
+      uint32_t value;
+      utils::ValueParser(string_value).parseUInt32(value).parseEnd();
+      ref = value;
+    } catch(const utils::ParseException&) {
+      return false;
+    }
```

Comment:
To me, the changes look like they are not more general, just moving the logic elsewhere, with an increase of required boilerplate. Not sure if this is an improvement, but maybe I'm missing a point.

Reply:
they are not ment to be more general, they are ment to be correct, all these conversion methods are incorrect as they all happily convert `"1banana"` to `1`